swim/member.go

```diff
@@ -184,14 +197,24 @@ func shouldProcessGossip(old *Member, gossip *Member) bool {
 		return false
 	}
 
-	// Only pick the gossip if the labels have a lower checksum then the checksum
-	// of the currently known labels
+	// keep the checksum values in local variables. The checksums are not cached
+	// and require some compute to get them, better to do once than twice.
 	gossipLabelsChecksum := gossip.Labels.checksum()
 	oldLabelsChecksum := old.Labels.checksum()
-	if gossipLabelsChecksum < oldLabelsChecksum {
+
+	// Gossips with a higher checksum should be processed to let the cluster
+	// converge to the labels that cause the highest checksum.
+	if gossipLabelsChecksum > oldLabelsChecksum {
```

Comment:
awesome!

Do you think we should add warn-logging here? different label(-checksums) with the same incarnation number is a sign of an implementation error.